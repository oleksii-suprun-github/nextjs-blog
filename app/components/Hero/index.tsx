import Link from 'next/link';

function Hero() {
  const href = '/publications';

  return (
    <div className="flex items-center justify-center px-10 pb-24 pt-24 text-stone-200">
      <div className="mx-auto max-w-screen-md">
        <span className="mb-4 flex text-5xl font-bold text-gray-200">NextJS Blog Â©</span>
        <h1 className="mb-6 break-normal text-4xl font-black leading-relaxed sm:text-6xl sm:leading-[1.25]">
          NextJS Blog Demo Project
        </h1>
        <p className="mb-8 text-xl leading-relaxed text-stone-200 sm:text-2xl sm:leading-[1.65]">
          This demo application, built with NextJS, TypeScript, Tailwind & Sanity allows users to
          check blog posts.
        </p>
        <div>
          <Link href={href}>
            <button className="btn no-animation btn-lg mb-5 border-0 bg-brand-pink text-gray-900 hover:bg-brand-light-pink">
              Get Started
            </button>
          </Link>
        </div>
      </div>
    </div>
  );
}

export default Hero;
