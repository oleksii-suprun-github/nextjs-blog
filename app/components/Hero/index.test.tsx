import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import Hero from '.';

vi.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    const { src, alt, width, height, ...rest } = props;
    return (
      // Use a simple img tag for testing purposes
      <img src={src} alt={alt} width={width} height={height} {...rest} />
    );
  },
}));

vi.mock('next/link', () => ({
  __esModule: true,
  default: ({ href, children }: any) => <a href={href}>{children}</a>,
}));

describe('Hero Component', () => {
  it('should render correctly and match snapshot', () => {
    const { asFragment } = render(<Hero />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the heading and description', () => {
    render(<Hero />);
    expect(
      screen.getByRole('heading', {
        name: /NextJS Newsletter Demo Project/i,
      }),
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        /This demo application, built with NextJS, TypeScript & Tailwind allows users to check latest company`s news and events.\./i,
      ),
    ).toBeInTheDocument();
  });

  it('should have a Get Started button that links to /publications', () => {
    render(<Hero />);
    const link = screen.getByRole('link', { name: /Get Started/i });
    expect(link).toHaveAttribute('href', '/publications');
  });
});
