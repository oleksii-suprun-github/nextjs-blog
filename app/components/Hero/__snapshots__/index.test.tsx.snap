// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero Component > should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center px-10 pb-24 pt-24 text-stone-200"
  >
    <div
      class="mx-auto max-w-screen-md"
    >
      <img
        alt="Logo"
        height="50"
        src="/logo.svg"
        width="150"
      />
      <h1
        class="mb-6 break-normal text-4xl font-black leading-relaxed sm:text-6xl sm:leading-[1.25]"
      >
        NextJS Newsletter Demo Project
      </h1>
      <p
        class="mb-8 text-xl leading-relaxed text-stone-200 sm:text-2xl sm:leading-[1.65]"
      >
        This demo application, built with NextJS, TypeScript & Tailwind allows users to check latest company\`s news and events.
      </p>
      <div>
        <a
          href="/publications"
        >
          <button
            class="btn btn-lg mb-5 border-0 bg-brand-pink text-gray-900 hover:bg-brand-light-pink"
          >
            Get Started
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
